### PR TITLE
Adds a test to verify correct generation of unique compound indexes for mssql

### PR DESCRIPTION
**Pull request by Shardj**
Creating this PR as requested on issue #3671

### CODE REVIEW

File: tests/Doctrine/Tests/DBAL/Platforms/AbstractSQLServerPlatformTestCase.php [hunk 1]
```diff
@@ -1586,4 +1587,53 @@ private function expectCteWithMinAndMaxRowNums(string $expectedSql, int $expecte
...
+                )
+            ]
+        );
+        $schema = new Schema([$table]);
+        $queries = $schema->toSql($this->platform);
+
+        self::assertSame(
+            [
+                'CREATE TABLE foo (id INT IDENTITY NOT NULL, workspace_id INT, search_engine_id INT, PRIMARY KEY (id))',
+                'CREATE UNIQUE INDEX key_COMPOUND ON foo (idA, idB)'
```
[the remark this answers is not in the document]
Comment (Shardj) [reply tests/Doctrine/Tests/DBAL/Platforms/AbstractSQLServerPlatformTestCase.php]: Sorry but I don't know what you mean, could you dumb it down for me. I'm just providing this test to demonstrate the issue as a runnable example as was requested on issue #3671

[the remark this answers is not in the document]
Comment (SenseException) [reply tests/Doctrine/Tests/DBAL/Platforms/AbstractSQLServerPlatformTestCase.php]: Please take a look at what @morozov wrote in the issue: https://github.com/doctrine/dbal/issues/3671#issuecomment-531406602. He linked there to an example test class.
